### ISSUE TITLE
Remove deprecated classnames from Navigation Link block

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#35237]
## What problem does this address?

After #33918 and #34171, there are a some extra classnames on Navigation Link block that are no longer needed. It would be good to remove them before the block lands in core.

List of classnames we can remove: 

* `wp-block-navigation-link__container`
* `wp-block-navigation-link__content` (one of the e2e tests is using this one as a selector so we should also update that)
* `wp-block-navigation-link__submenu-icon`
